### PR TITLE
tld: Use Wiki API

### PR DESCRIPTION
**Pull request by half-duplex**
### Description
Use the Wikipedia API to fetch only [article content](https://en.wikipedia.org/wiki/Special:ApiSandbox#action=parse&format=json&page=List_of_Internet_top-level_domains&prop=text&utf8=1&formatversion=2) instead of scraping the *entire* page.
I'd like to use a better parser, but we probably don't want to add dependencies to the larger sopel repo just for this.
`make qa` currently fails in the same way it does on `master`.

### Checklist
- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

**Comment by half-duplex** [comment]
Ok, I dunno why qa fails on my box, but it passes in CI :shrug:

**Comment by dgw** [comment]
> `make qa` currently fails in the same way it does on `master`.

😕

> Ok, I dunno why qa fails on my box, but it passes in CI 🤷

Good; I was about to say `master` doesn't fail anything so WTF 😅

File: sopel/modules/tld.py [hunk 1]
```diff
@@ -234,8 +234,20 @@ def _update_tld_data(bot, which):
...
+                    "action": "parse",
+                    "format": "json",
+                    "prop": "text",
+                    "utf8": 1,
+                    "formatversion": 2,
+                    "page": WIKI_PAGE_NAME,
+                },
+            ).json()
+            tld_data = tld_response["parse"]["text"]
+        except Exception:
```
Comment (dgw) [inline sopel/modules/tld.py:250]: Don't really like that change. Any exception that isn't from Requests is very unexpected and should fail loudly instead of just logging a warning.

I guess in addition to `RequestException`s, you could also catch ~~`JSONDecodeError`~~ (not available in py2) `ValueError` just in case decoding the JSON fails for some reason? But any other kind of error really shouldn't get swept under the `except Exception` rug. You'll find a number of `# TODO: Be specific` comments throughout Sopel's code because we're trying to eliminate this pattern wherever possible.

**Review by dgw — CHANGES_REQUESTED** [review]
tl;dr: `except Exception` bad

File: sopel/modules/tld.py (hunk 1 above)
Comment (half-duplex) [reply sopel/modules/tld.py:250]: I can't think of anything but syntax errors, requests errors, JSONDecodeError, or KeyError. What would cause ValueError?
Still, fair.

Comment (dgw) [reply sopel/modules/tld.py]: In older Pythons, `json` throws `ValueError` if decoding the JSON document fails. In newer Pythons, it throws `JSONDecodeError`—a subclass of `ValueError`. So, catching `ValueError` is valid to catch failed JSON decoding across all Python versions Sopel supports in 7.x.

File: sopel/modules/tld.py [hunk 2]
```diff
@@ -234,8 +234,21 @@ def _update_tld_data(bot, which):
...
+                    "action": "parse",
+                    "format": "json",
+                    "prop": "text",
+                    "utf8": 1,
+                    "formatversion": 2,
+                    "page": WIKI_PAGE_NAME,
+                },
+            ).json()
+            tld_data = tld_response["parse"]["text"]
+        # py3.3 needs ValueError instead of more specific json.decoder.JSONDecodeError
```
Comment (dgw) [inline sopel/modules/tld.py:250]: It's all py<3.5, not only 3.3

Comment (half-duplex) [reply sopel/modules/tld.py:250]: frick

**Review by dgw — APPROVED** [review]
Man, I got distracted by the whole Travis CI build queue thing… This is a meaningful reduction in the amount of HTML the bot needs to transfer and parse. Why not?

**Review by Exirel — APPROVED** [review]
